iroh/src/client/blobs/batch.rs

```diff
@@ -0,0 +1,460 @@
+    pub async fn add_dir(&self, root: PathBuf) -> Result<TempTag> {
+        self.add_dir_with_opts(root, Default::default()).await
+    }
+
+    /// Write a blob by passing an async reader.
+    ///
+    /// This will use a default chunk size of 64KB, and a format of [BlobFormat::Raw].
+    pub async fn add_reader(
+        &self,
+        reader: impl AsyncRead + Unpin + Send + 'static,
+    ) -> anyhow::Result<TempTag> {
+        self.add_reader_with_opts(reader, Default::default()).await
+    }
+
+    /// Write a blob by passing a stream of bytes.
+    pub async fn add_stream(
+        &self,
+        input: impl Stream<Item = io::Result<Bytes>> + Send + Unpin + 'static,
+    ) -> Result<TempTag> {
+        self.add_stream_with_opts(input, Default::default()).await
+    }
+
+    /// Create a temp tag to protect some content (blob or hashseq) from being deleted.
+    ///
+    /// A typical use case is that you are downloading some data and want to protect it
+    /// from deletion while the download is ongoing, but don't want to protect it permanently
+    /// until the download is completed.
+    pub async fn temp_tag(&self, content: HashAndFormat) -> Result<TempTag> {
+        // Notify the server that we want one temp tag for the given content
+        self.0
+            .rpc
+            .rpc(BatchCreateTempTagRequest {
+                batch: self.0.batch,
+                content,
+            })
+            .await??;
+        // Only after success of the above call, we can create the corresponding local temp tag
+        Ok(self.local_temp_tag(content, None))
+    }
+
+    /// Write a blob by passing an async reader.
+    ///
+    /// This produces a stream from the reader with a hardcoded buffer size of 64KB.
+    pub async fn add_reader_with_opts(
+        &self,
+        reader: impl AsyncRead + Unpin + Send + 'static,
+        opts: AddReaderOpts,
+    ) -> anyhow::Result<TempTag> {
+        let AddReaderOpts { format, chunk_size } = opts;
+        let input = ReaderStream::with_capacity(reader, chunk_size);
+        self.add_stream_with_opts(input, format).await
+    }
+
+    /// Write a blob by passing bytes.
+    pub async fn add_bytes_with_opts(
+        &self,
+        bytes: impl Into<Bytes>,
+        format: BlobFormat,
+    ) -> Result<TempTag> {
+        let input = futures_lite::stream::once(Ok(bytes.into()));
+        self.add_stream_with_opts(input, format).await
+    }
+
+    /// Import a blob from a filesystem path.
+    ///
+    /// `path` should be an absolute path valid for the file system on which
+    /// the node runs, which refers to a file.
+    ///
+    /// If you use [ImportMode::TryReference], Iroh will assume that the data will not
+    /// change and will share it in place without copying to the Iroh data directory
+    /// if appropriate. However, for tiny files, Iroh will copy the data.
+    ///
+    /// If you use [ImportMode::Copy], Iroh will always copy the data.
```

Comment:
```suggestion
    /// If you use [`ImportMode::TryReference`], Iroh will assume that the data will not
    /// change and will share it in place without copying to the Iroh data directory
    /// if appropriate. However, for tiny files, Iroh will copy the data.
    ///
    /// If you use [`ImportMode::Copy`], Iroh will always copy the data.
```

I'm just a little bit of a fan of this kind of formatting :grin: